testnet/stacks-node/src/config.rs

```diff
@@ -1010,6 +1023,155 @@ pub struct NodeConfig {
+        CostEstimatorName::NaivePessimistic
+    }
+}
+
+impl Default for FeeEstimatorName {
+    fn default() -> Self {
+        FeeEstimatorName::ScalarFeeRate
+    }
+}
+
+impl Default for CostMetricName {
+    fn default() -> Self {
+        CostMetricName::ProportionDotProduct
+    }
+}
+
+impl CostEstimatorName {
+    fn panic_parse(s: String) -> CostEstimatorName {
+        if &s.to_lowercase() == "naive_pessimistic" {
+            CostEstimatorName::NaivePessimistic
+        } else {
+            panic!(
+                "Bad cost estimator name supplied in configuration file: {}",
+                s
+            );
+        }
+    }
+}
+
+impl FeeEstimatorName {
+    fn panic_parse(s: String) -> FeeEstimatorName {
+        if &s.to_lowercase() == "scalar_fee_rate" {
+            FeeEstimatorName::ScalarFeeRate
+        } else {
+            panic!(
+                "Bad fee estimator name supplied in configuration file: {}",
+                s
+            );
+        }
+    }
+}
+
+impl CostMetricName {
+    fn panic_parse(s: String) -> CostMetricName {
+        if &s.to_lowercase() == "proportion_dot_product" {
+            CostMetricName::ProportionDotProduct
+        } else {
+            panic!("Bad cost metric name supplied in configuration file: {}", s);
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct FeeEstimationConfig {
+    pub cost_estimator: Option<CostEstimatorName>,
+    pub fee_estimator: Option<FeeEstimatorName>,
+    pub cost_metric: Option<CostMetricName>,
+    pub log_error: bool,
+}
+
+impl Default for FeeEstimationConfig {
+    fn default() -> Self {
+        Self {
+            cost_estimator: Some(CostEstimatorName::default()),
+            fee_estimator: Some(FeeEstimatorName::default()),
+            cost_metric: Some(CostMetricName::default()),
+            log_error: true,
+        }
+    }
+}
+
+impl From<FeeEstimationConfigFile> for FeeEstimationConfig {
+    fn from(f: FeeEstimationConfigFile) -> Self {
+        if let Some(true) = f.disabled {
+            return Self {
+                cost_estimator: None,
+                fee_estimator: None,
+                cost_metric: None,
+                log_error: false,
+            };
+        }
+        let cost_estimator = f
+            .cost_estimator
+            .map(CostEstimatorName::panic_parse)
+            .unwrap_or_default();
+        let fee_estimator = f
+            .fee_estimator
+            .map(FeeEstimatorName::panic_parse)
+            .unwrap_or_default();
+        let cost_metric = f
+            .cost_metric
+            .map(CostMetricName::panic_parse)
+            .unwrap_or_default();
+        let log_error = f.log_error.unwrap_or(true);
+        Self {
+            cost_estimator: Some(cost_estimator),
+            fee_estimator: Some(fee_estimator),
+            cost_metric: Some(cost_metric),
+            log_error,
+        }
+    }
+}
+
+impl FeeEstimationConfig {
+    pub fn make_pessimistic_cost_estimator(
```

Comment:
Do we want to maybe put the cost and fee estimator state into the same database?  Or is there some sort of pathological locking behavior we're trying to avoid here?  Seems like the state isn't that big in both these databases.

Reply:
Just trying to avoid dependence between the two if it isn't necessary -- there aren't any queries that use both databases.